[Bug]: TestMetaCache_ExpireShardLeaderCache is not stable

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: b3287ca
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

TestMetaCache_ExpireShardLeaderCache is not stable

### Expected Behavior

Unit test passes

### Steps To Reproduce

_No response_

### Milvus Log

```
2023-03-09T03:38:45.2953528Z     meta_cache_test.go:837: 
2023-03-09T03:38:45.2954239Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/proxy/meta_cache_test.go:837
2023-03-09T03:38:45.2954772Z         	            				/go/src/github.com/milvus-io/milvus/internal/proxy/asm_amd64.s:1571
2023-03-09T03:38:45.2955170Z         	Error:      	"[<NodeID: 3> <NodeID: 1> <NodeID: 2>]" should have 2 item(s), but has 3
2023-03-09T03:38:45.2955539Z         	Test:       	TestMetaCache_ExpireShardLeaderCache
2023-03-09T03:38:46.2953049Z     meta_cache_test.go:837: 
2023-03-09T03:38:46.2953776Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/proxy/meta_cache_test.go:837
2023-03-09T03:38:46.2954307Z         	            				/go/src/github.com/milvus-io/milvus/internal/proxy/asm_amd64.s:1571
2023-03-09T03:38:46.2955005Z         	Error:      	"[<NodeID: 1> <NodeID: 2> <NodeID: 3>]" should have 2 item(s), but has 3
2023-03-09T03:38:46.2955373Z         	Test:       	TestMetaCache_ExpireShardLeaderCache
2023-03-09T03:38:47.2952220Z     meta_cache_test.go:834: 
2023-03-09T03:38:47.2952920Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/proxy/meta_cache_test.go:834
2023-03-09T03:38:47.2953297Z         	Error:      	Condition never satisfied
2023-03-09T03:38:47.2953626Z         	Test:       	TestMetaCache_ExpireShardLeaderCache
2023-03-09T03:38:49.2975576Z     meta_cache_test.go:880: 
2023-03-09T03:38:49.2976295Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/proxy/meta_cache_test.go:880
2023-03-09T03:38:49.2976819Z         	            				/go/src/github.com/milvus-io/milvus/internal/proxy/asm_amd64.s:1571
2023-03-09T03:38:49.2977186Z         	Error:      	"[]" should have 3 item(s), but has 0
2023-03-09T03:38:49.2977769Z         	Test:       	TestMetaCache_ExpireShardLeaderCache
2023-03-09T03:38:50.2974015Z     meta_cache_test.go:880: 
2023-03-09T03:38:50.2974726Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/proxy/meta_cache_test.go:880
2023-03-09T03:38:50.2975240Z         	            				/go/src/github.com/milvus-io/milvus/internal/proxy/asm_amd64.s:1571
2023-03-09T03:38:50.2975601Z         	Error:      	"[]" should have 3 item(s), but has 0
2023-03-09T03:38:50.2975938Z         	Test:       	TestMetaCache_ExpireShardLeaderCache
2023-03-09T03:38:51.2980378Z     meta_cache_test.go:877: 
2023-03-09T03:38:51.2981104Z         	Error Trace:	/go/src/github.com/milvus-io/milvus/internal/proxy/meta_cache_test.go:877
2023-03-09T03:38:51.2981507Z         	Error:      	Condition never satisfied
2023-03-09T03:38:51.2981856Z         	Test:       	TestMetaCache_ExpireShardLeaderCache
2023-03-09T03:38:51.2982308Z --- FAIL: TestMetaCache_ExpireShardLeaderCache (7.00s)
```

### Anything else?

_No response_

--- a/internal/proxy/meta_cache_test.go
+++ b/internal/proxy/meta_cache_test.go
@@ -834,7 +834,7 @@ func TestMetaCache_ExpireShardLeaderCache(t *testing.T) {
 	assert.Eventually(t, func() bool {
 		nodeInfos, err := globalMetaCache.GetShards(ctx, true, "collection1")
 		assert.NoError(t, err)
-		return assert.Len(t, nodeInfos["channel-1"], 2)
+		return len(nodeInfos["channel-1"]) == 2
 	}, 3*time.Second, 1*time.Second)
 
 	queryCoord.EXPECT().GetShardLeaders(mock.Anything, mock.Anything).Return(&querypb.GetShardLeadersResponse{
@@ -877,6 +877,6 @@ func TestMetaCache_ExpireShardLeaderCache(t *testing.T) {
 	assert.Eventually(t, func() bool {
 		nodeInfos, err := globalMetaCache.GetShards(ctx, true, "collection1")
 		assert.NoError(t, err)
-		return assert.Len(t, nodeInfos["channel-1"], 3) && assert.Len(t, nodeInfos["channel-2"], 3)
+		return len(nodeInfos["channel-1"]) == 3 && len(nodeInfos["channel-2"]) == 3
 	}, 3*time.Second, 1*time.Second)
 }

